frontend/packages/console-shared/src/utils/pod-utils.ts

```diff
@@ -302,3 +304,87 @@ export class TransformPodData {
     return dcPodsData;
   }
 }
+
+const getOwnedResources = <T extends K8sResourceKind>(
+  { metadata: { uid } }: K8sResourceKind,
+  resources: T[],
+): T[] => {
+  return _.filter(resources, ({ metadata: { ownerReferences } }) => {
+    return _.some(ownerReferences, {
+      uid,
+      controller: true,
+    });
+  });
+};
+
+const getPodData = (
+  data: PodData,
+): { inProgressDeploymentData: Pod[] | null; completedDeploymentData: Pod[] | null } => {
+  // Scaling no. of pods
+  if (data.currentRCPhase === 'Complete' && data.currentAvailableReplicas !== data.totalReplicas) {
+    return { inProgressDeploymentData: null, completedDeploymentData: data.current };
+  }
+
+  // Deploy
+  switch (data.strategy) {
+    case 'Recreate':
```

Comment:
It's better to use if statement here if there are only two conditions.

Reply:
fixed in https://github.com/openshift/console/pull/2456/commits/d069d28b40bb578d9a26810bb61269647f5d241b